### DISCUSSION TITLE
Is v1 desupported?

### DISCUSSION THREAD

**Discussion by LunaticMuch** [mdx-js/mdx#2150] [Q&A] [answered]
It might be I did not find it, but is the v1 desupported? Many libraries still use v1 and I've found a security issue with `trim`

```
 @docusaurus/core@2.1.0
  └─┬ @docusaurus/mdx-loader@2.1.0
    └─┬ @mdx-js/mdx@1.6.22
      └─┬ remark-parse@8.0.3
        └── trim@0.0.1
```

An update to remark-parse@9.0.0 should quickly solve the trouble.

**Comment by ChristianMurphy** [comment, marked as answer]
Please see any of the many times this question has been asked https://github.com/mdx-js/mdx/issues?q=is%3Aissue+trim+is%3Aclosed+vulnerable and discussed https://github.com/mdx-js/mdx/discussions?discussions_q=trim+security

To reiterate:
1. _This is not an exploit_, it is a potential slow down. remark-parse 9+, react-markdown 6+, and mdx 2 address this, and provide other performance improvements.
https://overreacted.io/npm-audit-broken-by-design provides some additional insights into why `npm audit` and `snyk`, while useful, can also be broken for packages like react and mdx, flagging non-issues.
2. MDX version 1 cannot be patched (https://github.com/mdx-js/mdx/issues/1548#issuecomment-838498279)
3. MDX version 2 no longer uses `trim` as a dependency
